### ISSUE TITLE
Using Git-LFS on both Windows and *nix

### ISSUE THREAD

**Issue by rajaditya-m** [git-lfs/git-lfs#2486]
I have a repository which is accessed from both Windows as well as *nix which uses Git-LFS to track large files (of MP4 type)
Once I used the repo to do some pull from a *nix machine. Since then 
1) Any operation on the repo (git-clone, git-pull) hangs on Windows machines (any machine - I tested with 3 machines with Windows 8 and Windows 10)
2) Everything runs fine on *nix systems (both Ubuntu and Mac OS) 

Is this a known issue and is there any way of undoing things?

**Comment by ttaylorr** [comment]
Hi @rajaditya-m thanks for opening this issue -- sorry that you're having trouble. I'm not immediately sure what might be causing the hang, but we could narrow it down with `GIT_TRACE=1`.

Could you run `GIT_TRACE=1 git pull` from inside your repository on the Windows machine and post the output here? That should help narrow things down, and we can go from there.

**Comment by rajaditya-m** [comment]
Hi @ttaylorr 
Thanks so much for your prompt response. Here is the relevant trace output

`E:\RunningWater [master ↓4]> git pull origin master
21:49:41.354514 git.c:371               trace: built-in: git 'pull' 'origin' 'master'
21:49:41.357528 run-command.c:350       trace: run_command: 'fetch' '--update-head-ok' 'origin' 'master'
21:49:41.388537 git.c:371               trace: built-in: git 'fetch' '--update-head-ok' 'origin' 'master'
21:49:41.395551 run-command.c:350       trace: run_command: 'git-remote-https' 'origin' 'https://github.com/rajaditya-m/
RunningWater.git'
21:49:41.997268 run-command.c:350       trace: run_command: 'github --credentials get'
21:49:43.719651 run-command.c:350       trace: run_command: 'github --credentials store'
21:49:45.078659 run-command.c:350       trace: run_command: 'rev-list' '--objects' '--stdin' '--not' '--all' '--quiet'
21:49:45.124916 run-command.c:350       trace: run_command: 'rev-list' '--objects' '--stdin' '--not' '--all' '--quiet'
21:49:45.154235 git.c:371               trace: built-in: git 'rev-list' '--objects' '--stdin' '--not' '--all' '--quiet'
21:49:45.185256 run-command.c:1130      run_processes_parallel: preparing to run up to 1 tasks
21:49:45.185256 run-command.c:1162      run_processes_parallel: done
21:49:45.185256 run-command.c:350       trace: run_command: 'gc' '--auto'
From https://github.com/rajaditya-m/RunningWater
 * branch            master     -> FETCH_HEAD
21:49:45.214103 git.c:371               trace: built-in: git 'gc' '--auto'
21:49:45.225110 run-command.c:350       trace: run_command: 'merge' 'FETCH_HEAD'
21:49:45.257638 git.c:371               trace: built-in: git 'merge' 'FETCH_HEAD'
Updating 56fea4e..218223f
21:49:45.633891 run-command.c:350       trace: run_command: 'git-lfs filter-process'
trace git-lfs: run_command: 'git' version
trace git-lfs: run_command: 'git' config -l
trace git-lfs: Initialize filter-process
trace git-lfs: Read filter-process request.`

At this point, no further activity is observed and the command hangs as it is.

**Comment by ttaylorr** [comment]
Hi @rajaditya-m, thanks for posting that output -- this looks like it could be a bug with the `git lfs filter-process` command blocking on reading input from Git.

I know that there was a bug like this in a previous version of LFS that we have since fixed. (See: #1932). Can you post the output of `git lfs version` and `git --version` to double check?

**Comment by rajaditya-m** [comment]
Here is the relevant output.

```
E:\RunningWater [master ↓4 +0 ~0 -1 !]> git version
14:01:22.915594 git.c:371               trace: built-in: git 'version'
git version 2.11.0.windows.3
E:\RunningWater [master ↓4 +0 ~0 -1 !]> git lfs version
14:01:33.314332 git.c:607               trace: exec: 'git-lfs' 'version'
14:01:33.315333 run-command.c:350       trace: run_command: 'git-lfs' 'version'
git-lfs/1.5.5 (GitHub; windows amd64; go 1.7.4; git c2dcd6f5)
```

I saw that my github client has already been updated to the latest version.

**Comment by ttaylorr** [comment]
Thanks for posting -- I think we fixed this bug in 2.0.0, so upgrading to the latest version (v2.2.1) should fix this right up.
